src/client/AccountsClient.js

```diff
@@ -50,11 +50,11 @@ class Accounts extends AccountsCommon {
   }
 
   // TODO Accept 'profile' in the options
-  async createUser(user: UserCreationInputType, callback: Function): Promise<void> {
+  async createUser(user: UserCreationInputType, callback: ?Function): Promise<void> {
     this.validatePassword(user.password);
     // TODO Throw error if client user creation is disabled
 
-    if (!this.validateUsername(user.username, false) && !this.validateEmail(user.email, false)) {
+    if (this.validateUsername(user.username, false) && this.validateEmail(user.email, false)) {
```

Comment:
The validateXXX methods should return true whenever email is valid and not when there's an error in my opinion. If change that, the logic you had before will be correct.